synapse/handlers/federation.py

```diff
@@ -1466,6 +1485,7 @@ async def _sync_partial_state_room(
                 `destination` is unavailable
             room_id: room to be resynced
         """
+        assert not self.config.worker.worker_app
```

Comment:
This is implied by the identical assert in `do_invite_join`, which starts the background task.